security/ssh_certificate.ksy

```diff
@@ -0,0 +1,159 @@
+      - id: valid_after
+        doc: Unix timestamp (seconds since 1970-01-01 00:00:00)
+        type: u8
+      - id: valid_before
+        doc: |
+          Unix timestamp (seconds since 1970-01-01 00:00:00)
+          When "forever" is requested this is set to 0xFFFF_FFFF_FFFF_FFFF
+        type: u8
+      - id: critical_options
+        doc: |
+          Contains zero or more options that are considered "critical".
+          They are considered "critical" as implementations must refuse
+          to authorize a certificate that has unrecognized options. This
+          prevents an unknown restriction in the certificate from failing to
+          be applied.
+        type: packed_cstring_tuple
+      - id: extensions
+        doc: |
+          Contains zero or more optional extensions. These extensions
+          are not critical, and an implementation that encounters one
+          that it does not recognize may safely ignore it.
+        type: packed_cstring_tuple
+      - id: reserved
+        doc: Unused currently
+        type: cstring_bytes
+      - id: signature_key
+        doc: |
+          The signature key field contains the CA key used to sign the
+          certificate. The valid key types for CA keys are ssh-rsa,
+          ssh-dss, ssh-ed25519 and the ECDSA types ecdsa-sha2-nistp256,
+          ecdsa-sha2-nistp384, ecdsa-sha2-nistp521. "Chained" certificates, where
+          the signature key type is a certificate type itself are NOT supported.
+          Note that it is possible for a RSA certificate key to be signed by a
+          Ed25519 or ECDSA CA key and vice-versa.
+        type: cstring_sshkey
+      - id: signature
+        doc: |
+          signature is computed over all preceding fields from the initial string
+          up to, and including the signature key. Signatures are computed and
+          encoded according to the rules defined for the CA's public key algorithm
+          (RFC4253 section 6.6 for ssh-rsa and ssh-dss, RFC5656 for the ECDSA
+          types), and draft-josefsson-eddsa-ed25519-03 for Ed25519.
+        type: packed_cstring_tuple
+    enums:
+      cert_type:
+        1: user
+        2: host
+  packed_cstring:
+    seq:
+      - id: len_packed_cstring
+        type: u4
+      - id: packed_strings
+        size: len_packed_cstring
+        type: cstrings
+  packed_cstring_tuple:
+    seq:
+      - id: len_packed_cstring_tuple
+        type: u4
+      - id: packed_strings
+        type: cstring_tuples
+        size: len_packed_cstring_tuple
+  cstrings:
+    seq:
+      - id: strings
+        type: cstring_utf8
+        repeat: eos
+  cstring_tuples:
+    seq:
+      - id: strings
+        type: cstring_tuple
+        repeat: eos
+  cstring_tuple:
+    seq:
+      - id: name
+        type: cstring_utf8
+      - id: data
+        type: cstring_bytes
+  cstring_sshkey:
+    seq:
+      - id: len_cstring_sshkey
+        type: u4
+      - id: value
+        type: ssh_public_key
```

Comment:
I'd go just for `len_value`:

```suggestion
      - id: len_value
        type: u4
      - id: value
        type: ssh_public_key
```

You don't need to specify that it relates to the `cstring_sshkey`, that's the name of the enclosing type.